src/properties.jl

```diff
@@ -246,39 +246,42 @@ rmatrices = Dict(
 
 #---------------------- Equivariant matrices
 
-struct EuclideanMatrix{T} <: AbstractProperty where T<:Real
-   val::SMatrix{3, 3, Complex{T}, 9}
+struct EuclideanMatrix{T} <: AbstractProperty
+   val::SMatrix{3, 3, T, 9}
 end
 
 Base.show(io::IO, φ::EuclideanMatrix) = 
    print(io,  "m[$(φ.val[1,1]), $(φ.val[1,2]), $(φ.val[1,3])]
 m[$(φ.val[2,1]), $(φ.val[2,2]), $(φ.val[2,3])] 
 m[$(φ.val[3,1]), $(φ.val[3,2]), $(φ.val[3,3])]")
 
-real(φ::EuclideanMatrix) = EuclideanMatrix(φ.val)
-complex(φ::EuclideanMatrix) = EuclideanMatrix(φ.val)
+real(φ::EuclideanMatrix) = EuclideanMatrix(real.(φ.val))
+complex(φ::EuclideanMatrix) = EuclideanMatrix(complex(φ.val))
 complex(::Type{EuclideanMatrix{T}}) where {T} = EuclideanMatrix{complex(T)}
 
 +(x::SMatrix{3}, y::EuclideanMatrix) = EuclideanMatrix(x + y.val)
 Base.convert(::Type{SMatrix{3, 3, T, 9}}, φ::EuclideanMatrix) where {T} =  convert(SMatrix{3, 3, T, 9}, φ.val)
 
-isrealB(::EuclideanMatrix) = true
+isrealB(::EuclideanMatrix) where {T} = (T == real(T))
 isrealAA(::EuclideanMatrix) = false
 
 
 #fltype(::EuclideanMatrix{T}) where {T} = T
 
-EuclideanMatrix{T}() where {T <: Real} = EuclideanMatrix{T}(zero(SMatrix{3, 3, ComplexF64, 9}))
+EuclideanMatrix{T}() where {T <: Number} = EuclideanMatrix{T}(zero(SMatrix{3, 3, T, 9}))
 EuclideanMatrix(T::DataType=Float64) = EuclideanMatrix{T}()
 
 
-function filter(φ::EuclideanMatrix, grp::O3, b::Array)
-   if length(b) <= 1 #MS: Not sure if this should be here
+function filter(φ::EuclideanMatrix, grp::O3, bb::Array)
+   if length(bb) == 0  # no zero-correlations allowed 
+      return false 
+   end
+   if length(bb) == 1 #MS: Not sure if this should be here
```

Comment:
I don't remember the issue - same with EuclideanVector. If codes run ok, leave it for now but file an issue to investigate this. Do you have an idea?

Reply:
The code runs. Issues is filed!